ResearchKit/Common/ORKResultPredicate.h

```diff
@@ -508,20 +838,48 @@ ORK_CLASS_AVAILABLE
  
  @param taskIdentifier                  The identifier of the task whose result you want to match.
                                             Pass `nil` to match the ongoing task.
+ @param stepIdentifier                  The identifier of the step whose result you want to match.
+                                            Passing `nil` assumes the step identifier is the same as
+                                            the result identifier.
  @param resultIdentifier                The identifier of the question result you are interested in.
- @param minimumExpectedAnswerHour       The minimum expected hour component value.
- @param minimumExpectedAnswerMinute     The minimum expected minute component value.
- @param maximumExpectedAnswerHour       The maximum integer hour component value.
- @param maximumExpectedAnswerMinute     The maximum expected minute component value.
+ @param minimumExpectedHour             The minimum expected hour component value.
+ @param minimumExpectedMinute           The minimum expected minute component value.
+ @param maximumExpectedHour             The maximum integer hour component value.
+ @param maximumExpectedMinute           The maximum expected minute component value.
  
  @return A result predicate.
  */
 + (NSPredicate *)predicateForTimeOfDayQuestionResultWithTaskIdentifier:(nullable NSString *)taskIdentifier
+                                                        stepIdentifier:(nullable NSString *)stepIdentifier
                                                       resultIdentifier:(NSString *)resultIdentifier
-                                             minimumExpectedAnswerHour:(NSInteger)minimumExpectedAnswerHour
-                                           minimumExpectedAnswerMinute:(NSInteger)minimumExpectedAnswerMinute
-                                             maximumExpectedAnswerHour:(NSInteger)maximumExpectedAnswerHour
-                                           maximumExpectedAnswerMinute:(NSInteger)maximumExpectedAnswerMinute;
+                                                   minimumExpectedHour:(NSInteger)minimumExpectedHour
+                                                 minimumExpectedMinute:(NSInteger)minimumExpectedMinute
+                                                   maximumExpectedHour:(NSInteger)maximumExpectedHour
+                                                 maximumExpectedMinute:(NSInteger)maximumExpectedMinute;
+
```

Comment:
To be consistent with `ORKTimeOfDayQuestionResult`.
Can you use dateComponents as the values?